packages/generators/generators/lib/plops/migration.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const moment = require('moment');
+const tsUtils = require('@strapi/typescript-utils');
+
+
+const validateInput = (input) => {
+  const regex = /^[A-Za-z-_0-9]+$/g;
+
+  if (!input) {
+    return 'You must provide an input';
+  }
+
+  return regex.test(input) || "Please use only letters and number, '-' or '_' and no spaces";
+};
+
+module.exports = (plop) => {
+  // Migration generator
+  plop.setGenerator('migration', {
+    description: 'Generate a migration',
+    prompts: [
+      {
+        type: 'input',
+        name: 'id',
+        message: 'Migration name',
+        validate: (input) => validateInput(input),
+      },
+    ],
+    actions(answers) {
+      const currentDir = process.cwd();
+      const language = tsUtils.isUsingTypeScriptSync(currentDir) ? 'ts' : 'js';
+      const timestamp = moment().format('YYYY_MM_DD_HH_mm_ss');
```

Comment:
You can create a small util to format the date instead of using moment.
```js
const getFormattedDate = (date = new Date()) => {
  return new Date(date.getTime() - date.getTimezoneOffset() * 60000)
    .toISOString()
    .split(".")[0]
    .replace(/[-T:]/g, "_");
};
```
![image](https://user-images.githubusercontent.com/25851739/211769386-c23ab839-20f6-4f51-b249-352e31c4ab74.png)